### PR TITLE
net-mail/cyrus-imapd: new version 3.2.2, p/maint suggestion

### PR DESCRIPTION
**Pull request by RarogCmex**
New ebuild with version improvements and cleaning unused.
Cyrus-imapd now don't use lmdb, I don't know why.
Tests is always required now.
Also I suggest my proxy maintaining.
Closes:https://bugs.gentoo.org/688242
Closes:https://bugs.gentoo.org/687864
Closes:https://bugs.gentoo.org/662896
Closes:https://bugs.gentoo.org/356341
Closes:https://bugs.gentoo.org/721466

Package-Manager: Portage-2.3.99, Repoman-2.3.23
Signed-off-by: Denis Reva <denis7774@gmail.com>

### CODE REVIEW

**Comment by RarogCmex** [comment]
Closes: https://bugs.gentoo.org/731266

File: net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,208 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit autotools flag-o-matic pam ssl-cert
```
Comment (juippis) [inline net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild:5]: One empty line between EAPI and inherit please.

File: net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+SRC_URI="https://github.com/cyrusimap/${PN}/releases/download/${P}/${P}.tar.gz"
+
+LICENSE="BSD-with-attribution GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE="afs backup calalarm caps clamav http kerberos ldap \
+	mysql nntp pam perl postgres replication +server sieve snmp \
+	sqlite ssl static-libs tcpd xapian"
+
+CDEPEND="
```
Comment (juippis) [inline net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild:18]: Don't use `CDEPEND`, but something to describe it better. Like widely used `COMMON_DEPEND`. 
https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide/Style_Guide#Avoid_using_single-char_.2ADEPEND_variable

File: net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+	sys-libs/zlib:0=
+	dev-libs/libpcre:3
+	>=dev-libs/cyrus-sasl-2.1.17:2
+	dev-libs/jansson
+	dev-libs/icu:0=
+	sys-libs/e2fsprogs-libs
```
Comment (juippis) [inline net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild:24]: Sort this section properly.

File: net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+	for i in `grep -rl -e 'master\.8' -e 'master(8)' "${S}"` ; do
+		sed -i -e 's:master\.8:cyrusmaster.8:g' \
+			-e 's:master(8):cyrusmaster(8):g' \
+			"${i}" || die "sed failed" || die "sed failed"
+	done
+	mv man/master.8 man/cyrusmaster.8 || die "mv failed"
+	sed -i -e "s:MASTER:CYRUSMASTER:g" \
+		-e "s:Master:Cyrusmaster:g" \
+		-e "s:master:cyrusmaster:g" \
+		man/cyrusmaster.8 || die "sed failed"
```
Comment (juippis) [inline net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild:79]: This is hacky. Isn't there a better way? What good does the grep do here? You should know what files need sedding and what does not.

File: net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+		$(use_with perl) \
+		$(use_with sqlite) \
+		$(use_with ssl openssl) \
+		$(use_enable server) \
+		$(use_enable sieve) \
+		$(use_with snmp) \
+		$(use_enable static-libs static) \
+		$(use_with tcpd libwrap) \
+		$(use_enable xapian) \
+		${myconf}
```
Comment (juippis) [inline net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild:130]: Why not go for the array form to get rid of these backlashes?

File: net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+}
+
+src_install() {
+	emake DESTDIR="${D}" INSTALLDIRS=vendor install
+
+	dodoc README*
+	dodoc -r doc
+	cp -r contrib tools "${D}/usr/share/doc/${PF}"
```
Comment (juippis) [inline net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild:138]: Needs to be appended with `|| die`

File: net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+	rm -f doc/text/Makefile*
```
Comment (juippis) [inline net-mail/cyrus-imapd/cyrus-imapd-3.2.2.ebuild:139]: Don't use `-f`. Append with `|| die`. Do these still get installed with `dodoc` above?

File: net-mail/cyrus-imapd/metadata.xml [hunk 8]
```diff
@@ -1,7 +1,14 @@
...
 <pkgmetadata>
-  <!-- maintainer-needed -->
+  <maintainer type="person">
+   <email>denis7774@gmail.com</email>
+   <name>Reva Denis aka RarogCmex</name>
+  </maintainer>
+  <maintainer type="project">
+   <email>proxy-maint@gentoo.org</email>
+   <name>Proxy Maintainers</name>
+  </maintainer>
```
Comment (juippis) [inline net-mail/cyrus-imapd/metadata.xml:11]: Also fix the CI issue you introduced with this.

**Comment by RarogCmex** [comment]
@juippis I've switch to mail-mta/courier (both mta and imap) so I'm doubt should I support cyrus-imapd now? Can they work both?

**Comment by juippis** [comment]
I'm sorry, I don't understand the question. Is *cyrus-imapd* some reverse dependency that you thought you needed, but found out *courier* works better? Yeah, don't take on burden to maintain packages you don't use/need.

**Comment by RarogCmex** [comment]
@juippis Yes, I don't want maintain it anymore. Should I close the PR?

**Comment by juippis** [comment]
If you have no intention to fix above marks, yes please, close it. 

Remember that you can still make commits to maintainer-needed packages without becoming the maintainer, if you want to train / care about it for some reason.

**Comment by ajakk** [comment]
Seems like there's no interest in this anymore, but please remember you can still work on packages you don't maintain!

**Pull request closed without merging**